### DISCUSSION TITLE
Is getServerSideProps() safe?

### DISCUSSION THREAD

**Discussion by devalnor** [vercel/next.js#17357] [Help] [answered]
Hi,

In order to use an "secret" API. I would like to know if getServerSideProps() is not bundled and not visible in source code on the client side.

Someone can confirme that's is OK to use getServerSideProps() for secret API requests on the server side only.

Thx,
Mx

**Comment by jamesmosier** [comment, marked as answer]
`getServerSideProps` code never makes it to your users browsers. All code contained in that function is removed from any client side bundles. So yes, it is safe.

If you want to see what gets eliminated from client side bundles, places you code in this app and see what it outputs: https://next-code-elimination.now.sh/

**Reply by devalnor** [reply]
Thanks for the link very useful 👍
